### PR TITLE
Slight refactor of the normalization process

### PR DESCRIPTION
**Pull request by ravwojdyla**

### CODE REVIEW

File: tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala [hunk 1]
```diff
@@ -19,10 +19,12 @@ package com.spotify.featran
 
 import org.tensorflow.example.{Example, Features}
 import org.tensorflow.{example => tf}
+import _root_.java.util.regex.Pattern
 
 package object tensorflow {
+  private val featureNameNormalization = Pattern.compile("[^A-Za-z0-9_]")
```
Comment (richwhitjr) [inline tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: Patterns serializable or should this be lazy?  Although not quite clear to me at a brief glance if serialization would be attempted here.

Comment (ravwojdyla) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: @richwhitjr it's inside an object so we get serialization for "free".

Comment (regadas) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: Also Pattern is serializable.

Comment (regadas) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: @ravwojdyla 🎨 can we make it FeatureNameNormalization?

**Comment by ravwojdyla** [comment]
@regadas why?

File: tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala (hunk 1 above)
Comment (ravwojdyla) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: @regadas why?

Comment (regadas) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: to follow scala style convention on constant names.

Comment (ravwojdyla) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: Does this codebase follow that rule tho, see:
 * MultiFeatureExtractor
 * FlinkCollectionType
 * MDLPDiscretizer

?

It seems like the above is the most common here.

Comment (regadas) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: that's why I marked the comment with 🎨 and not requested changes 😄 it's up to you

Comment (ravwojdyla) [reply tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala]: 👍

**Pull request merged**